packages/plugins/src/plugins/aave/helpers.ts

```diff
@@ -11,77 +11,86 @@ import {
   Yield,
   formatTokenAddress,
   getElementLendingValues,
-  networks,
+  BorrowLendRate,
+  aprToApy,
+  TokenPriceSource,
+  TokenPrice,
+  EvmNetworkIdType,
 } from '@sonarwatch/portfolio-core';
 import {
   UiIncentiveDataProvider,
   UiPoolDataProvider,
-} from '@aave/contract-helpers';
-import { StaticJsonRpcProvider } from '@ethersproject/providers';
+} from '@aave/contract-helpers-v2';
 import {
-  UserIncentiveDict,
   formatUserSummaryAndIncentives,
-} from '@aave/math-utils';
+  UserIncentiveDict,
+} from '@aave/math-utils-v2';
+import { Address, ContractFunctionConfig } from 'viem';
 import {
+  FormattedReserve,
   LendingConfig,
   LendingData,
   ReserveYieldInfo,
   UserReserveData,
+  UserReserveDataV3,
   UserSummary,
 } from './types';
 import { Cache } from '../../Cache';
-import { getRpcEndpoint } from '../../utils/clients/constants';
-import { platformId } from './constants';
+import { lendingPoolsPrefix } from './constants';
+import getEvmEthersClient from '../../utils/clients/getEvmEthersClient';
+import { getEvmClient } from '../../utils/clients';
+import { balanceOfErc20ABI } from '../../utils/evm/erc20Abi';
+import { zeroBigInt } from '../../utils/misc/constants';
+import { stkAbi } from './abi';
+
+export function getUiProviders(
+  networkId: EvmNetworkIdType,
+  lendingConfig: LendingConfig
+) {
+  const provider = getEvmEthersClient(networkId);
+  const poolDataProvider = new UiPoolDataProvider({
+    uiPoolDataProviderAddress: lendingConfig.uiPoolDataProviderAddress,
+    provider,
+    chainId: lendingConfig.chainId,
+  });
 
-export const lendingPoolsPrefix = 'aave-lendingPools';
+  const incentiveDataProvider = new UiIncentiveDataProvider({
+    uiIncentiveDataProviderAddress:
+      lendingConfig.uiIncentiveDataProviderAddress,
+    provider,
+    chainId: lendingConfig.chainId,
+  });
+
+  return {
+    poolDataProvider,
+    incentiveDataProvider,
+  };
+}
 
 export async function fetchLendingForAddress(
   address: string,
-  networkId: NetworkIdType,
+  networkId: EvmNetworkIdType,
   configs: LendingConfig[],
   cache: Cache,
+  platformId: string,
   proxyInfo?: ProxyInfo
 ): Promise<PortfolioElement[]> {
-  const elements: PortfolioElementBorrowLend[] = [];
-  for (let i = 0; i < configs.length; i++) {
-    const config = configs[i];
-    if (!config) continue;
-    const {
-      lendingPoolAddressProvider,
-      uiPoolDataProviderAddress,
-      uiIncentiveDataProviderAddress,
-    } = config;
+  const elementPromises = configs.map(async (config) => {
+    if (!config) return null;
```

Comment:
Why force dealing with `null` in the rest of the code ? (as opposed to sticking to the previous version that assumed `undefined` -- which is more _Typescript-native_ ?)

```suggestion
    if (!config) return;
```

Reply:
previous version was a for loop. null signifies intent vs undefined meaning uninitialized/missing? I wasnt aware of undefined being more typescript native. 

I tried your suggestion but there is an eslint rule which doesnt allow for returning undefined with another value
https://typescript-eslint.io/rules/consistent-return